library/std/src/fs.rs

```diff
@@ -1406,7 +1406,7 @@ impl Permissions {
     /// On Unix-based platforms this checks if *any* of the owner, group or others
     /// write permission bits are set. It does not check if the current
     /// user is in the file's assigned group. It also does not check ACLs.
-    /// Therefore even if this returns true you may not be able to write to the
+    /// Therefore even if this returns false you may not be able to write to the
```

Comment:
Perhaps this should be

> Therefore even if this returns true you may not be able to read to the file

Or even

> Therefore you may not be able to rely on the return value of this function to predict the success of reads or writes on the file.

Reply:
I think the second approach is preferable; the first one is better than the original (since it is accurate), but changes the text from talking about writes to reads (while the focus of the `read_only` function is checking the write permissions).

I.e. any text should allow the reader to make the inference that:

1. if it returns `true` doesn't say anything about being actually able to read (all the UNIX read permissions could be unset)
2. if it returns `false` that doesn't mean *you* can write to the file